### ISSUE TITLE
ZipCacher - FhirArtifactCache location

### ISSUE THREAD

**Issue by tletz** [FirelyTeam/firely-net-sdk#2019]
We use the API to host a FHIR Service using ZipSource to have a source for the CachedResolver for the profile definitions.
ZipSource uses ZipCacher to extract these profile definitions to %TEMP% (via GetTempFilePath) - which is hardcoded.

**problem**
On production environments at customer side occurs a periodically cleanup of the %TEMP% directory which destroys the cache and will cause our service to reject all incomming resources due to validation not succeeded

Overall result: FAILURE (1 errors and 0 warnings) [ERROR] [ERROR] Resolution of profile at 'http://hl7.org/fhir/StructureDefinition/Condition'failed:EinTeildesPfades"C:\Windows\TEMP\FhirArtifactCache-3.5.0-Hl7.Fhir.R4.Specification\tmpEFD4\profiles-resources.xml" konnte nicht gefunden werden. (at Condition).

**solution** 

We would like to be able to specify the location of the artifact cache to set it to another directory, e.g. some folder beneath %programdata%. So a solution could be to have a property within DirectorySourceSettings which is passed to ZipCacher to specify the location of the artifact cache.

**Comment by pbach-tiplu** [comment]
We unfortunately encountered the same problem.

At customer side we have to run our application in IIS with the ApplicationPoolIdentity. The `C:\Windows\Temp` folder got cleaned up and our ZipSource crashed with an exception.

`System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Windows\TEMP\FhirArtifactCache-4.0.0+7749a7f393de9ccaa0585a71daea9a7c2fa07bfb-Hl7.Fhir.R4.Specification\specification\search-parameters.xml'`

The `ZipSource` had a change (https://github.com/FirelyTeam/firely-net-sdk/issues/1722) that the we can decide the location of the `specification.zip`. Would be great if we can get control over the cache folder aswell.

**Comment by ewoutkramer** [comment]
I am currently revising the ZipSource, will pick this up for version 5.0

**Comment by ewoutkramer** [comment]
Done.

**Comment by DanieleBini** [comment]
> Done.

Actually we use the 4.2.1 version, all component of our solution use the R4 version. 
It is possible to have this new functionality in the 4.x version?

**Comment by ewoutkramer** [comment]
No, the codebase has changed at lot going from 4.x-> 5.x (we totally reorganized the projects and the repo), so it's not easily merged back anymore.  But that should not be a problem, we support R4 fully in the SDK 5.x (and STU3 too and R4B). What's holding you back from upgrading to SDK 5?
